src/lib.rs

```diff
@@ -507,10 +512,11 @@ impl Regex {
         if lower == upper {
             regex.compiled.repeat_const *= upper;
         } else {
-            regex
-                .compiled
-                .repeat_ranges
-                .push(Uniform::new_inclusive(lower, upper).expect("upper >= lower"));
+            regex.compiled.repeat_ranges.push(
+                Uniform::new_inclusive(lower, upper)
+                    .ok()
+                    .ok_or(Error::Unsatisfiable)?,
```

Comment:
```suggestion
                    .map_err(|_| Error::Unsatisfiable)?,
```

Reply:
This is very much just a matter of taste, unless you think that the compiler will produce different results?

Reply:
@414owen  The `.ok().ok_or()` route only works for these context-less errors, like the current `Error::Unsatisfiable`. If we wish to incorporate details about the original error into the new error you must use `.map_err()`, calling `.ok()` will make the original error unavailable.